eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Utxo.hs

```diff
@@ -611,29 +610,36 @@ validateMaxTxSizeUTxO pp tx =
     txSize = tx ^. sizeTxF
 
 updateUTxOState ::
-  EraTxBody era =>
+  (EraTxBody era, Monad m) =>
   PParams era ->
   UTxOState era ->
   TxBody era ->
-  Coin ->
+  CertState era ->
   GovState era ->
-  UTxOState era
-updateUTxOState pp UTxOState {utxosUtxo, utxosDeposited, utxosFees, utxosStakeDistr, utxosDonation} txb depositChange govState =
-  let UTxO utxo = utxosUtxo
-      !utxoAdd = txouts txb -- These will be inserted into the UTxO
+  (SafeHash (EraCrypto era) EraIndependentTxBody -> Coin -> m ()) ->
```

Comment:
```suggestion
  (Coin -> m ()) ->
```
The TxBody hash seems redundant, since whatever's using this function already has access to the transaction body.

Reply:
Actually, I think it'd be better if `updateUTxOState` was split into two functions: one that computes `depositChange` and then another that updates the UTxOState. I'm just not a big fan of callbacks :sweat_smile:

Reply:
Done

Reply:
>  I'm just not a big fan of callbacks

I love callback